app/validators/questions/calculated_question_validator.py

```diff
@@ -46,19 +50,35 @@ def _validate_answer_is_numeric(self, answer_id):
                 referenced_answer=answer_id,
             )
 
-    def validate_calculations_value_source_is_numeric(self):
+    def validate_calculations_numeric_matching_answer_types(self):
         """
-        Validates that source answer is of number type
+        Validates that source answer is of number type, and that the answers_to_calculate match that type
         """
         for calculation in self.question.get("calculations"):
-            value = calculation.get("value")
+            if not (answer_id := calculation.get("answer_id")):
+                value = calculation.get("value")
+                # Calculated summary value source is validated elsewhere and must be of a number type
+                if isinstance(value, dict) and value.get("source") == "answers":
+                    answer_id = value.get("identifier")
 
-            if answer_id := calculation.get("answer_id"):
+            if answer_id:
                 self._validate_answer_is_numeric(answer_id)
+                self._validate_answers_are_same_numeric_type(
+                    calculation.get("answers_to_calculate"), answer_id
+                )
```

Comment:
This would mean `answers_to_calculate` isn't checked if `answer_id` isn't set.

We should move this outside the if and handle answer id not existing in the validate method

Reply:
Good spot - and we'd also have an issue if they aren't numeric and no referenced `answer_id`, moved both of these checks out of the if and passed them a dict of answer types instead + new test for this scenario